spec/forms/two_factor_options_form_spec.rb

```diff
@@ -71,22 +89,27 @@
       end
 
       it 'does not submit the phone when selected as the first single option' do
-        result = subject.submit(selection: ['phone'])
-
-        expect(result.success?).to eq false
+        expect(submit_phone.success?).to eq false
       end
     end
 
     context 'when a user wants to select phone as their second authentication method' do
-      let(:user) { create(:user, :with_authentication_app) }
+      let(:user) { build(:user, :with_authentication_app) }
+      let(:enabled_mfa_methods_count) { 1 }
+      let(:mfa_selection) { ['phone'] }
+
       before do
         allow(IdentityConfig.store).to receive(:select_multiple_mfa_options).and_return(true)
       end
 
       it 'submits the form' do
-        result = subject.submit(selection: ['phone'])
+        expect(submit_phone.success?).to eq true
+      end
 
-        expect(result.success?).to eq true
+      it 'includes analytics hash with a method count of one' do
+        expect(FormResponse).to receive(:new).with(hash_including(analytics_payload))
+
+        submit_phone
```

Comment:
ditto about not stubbing
```suggestion
        result = submit_phone
        expect(result.to_h).to eq(analytics_payload)
```